sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PreparePubsubWriteDoFn.java

```diff
@@ -175,6 +193,16 @@ public void process(
           .add("pubsub", "topic", PubsubClient.topicPathFromPath(topic).getDataCatalogSegments());
       reportedLineage = topic;
     }
+    // TODO: Remove this check once Dataflow's native sink supports ordering keys.
```

Comment:
should we remove this? It could break existing users where an ordering key is set (and ignored) but publishing of the message still occurs.  If we document that ordering key is only published if the withOrderingKey option is used then I think it would be better to drop the key here and logging that it is being dropped.

Reply:
@sjvanrossum wdyt? I raised the same concern in the other [thread](https://github.com/ahmedabu98/beam/pull/427#discussion_r1650304238) and am still leaning in that direction

Reply:
Seems good to me. My intention with being more strict about this was to prevent user confusion or potential data loss in case the ordering key can't be reconstructed from the message, but that disregards how the transform may be used today.

Adding this check would force users to scrub messages before output which I now realize does not look elegant in the flow of a pipeline.

Reply:
Thanks, just removed these lines as well as the comment in CHANGES.md